src/matrix/net/Reconnector.ts

```diff
@@ -66,8 +67,24 @@ export class Reconnector {
         return 0;
     }
 
+    get isStarted(): boolean {
+        return this._isStarted;
+    }
+
+    start(): void {
+        this._isStarted = true;
+    }
+
+    stop(): void {
+        this._isStarted = false;
+    }
+
     async onRequestFailed(hsApi: HomeServerApi): Promise<void> {
-        if (!this._isReconnecting) {  
+        if (!this._isStarted) {
+            return;
+        }
```

Comment:
`this._isStarted` is set to `false` on object construction, so will be false until `start()` is called. So this `onRequestFailed()` function will do nothing until `start()` is called, which is what we want, because we don't want to enter the reconnect loop if the `Reconnector` hasn't been started.

(I've just pushed a commit that explicitly sets `this._isStarted` to false in the constructor, but doesn't functionally change the behaviour, it was already being set to `false` on construction.)

Does that make it more clear?

Reply:
I guess I don't see whether `onRequestFailed()` can be invoked even if Reconnector hasn't started. If so, then all good.

Reply:
`onRequestFailed()` is called by `HomeserverApi` when a request fails :)